kubectl-fdb/cmd/evacuate.go

```diff
@@ -0,0 +1,172 @@
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+
+	"k8s.io/apimachinery/pkg/fields"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/FoundationDB/fdb-kubernetes-operator/controllers"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ctx "context"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+)
+
+func newEvacuateCmd(streams genericclioptions.IOStreams, rootCmd *cobra.Command) *cobra.Command {
+	o := NewFDBOptions(streams)
+	var nodeSelectors map[string]string
+
+	cmd := &cobra.Command{
+		Use:   "evacuate",
+		Short: "Adds all instance (or multiple) that run on a node to the remove list of the given cluster",
+		Long:  "Adds all instance (or multiple) that run on a node to the remove list of the given cluster",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			force, err := rootCmd.Flags().GetBool("force")
+			if err != nil {
+				return err
+			}
+			cluster, err := cmd.Flags().GetString("fdb-cluster")
+			if err != nil {
+				return err
+			}
+			withExclusion, err := cmd.Flags().GetBool("exclusion")
+			if err != nil {
+				return err
+			}
+			nodeSelector, err := cmd.Flags().GetStringToString("node-selector")
+			if err != nil {
+				return err
+			}
+			config, err := o.configFlags.ToRESTConfig()
+			if err != nil {
+				return err
+			}
+
+			scheme := runtime.NewScheme()
+			_ = clientgoscheme.AddToScheme(scheme)
+			_ = fdbtypes.AddToScheme(scheme)
+
+			kubeClient, err := client.New(config, client.Options{Scheme: scheme})
+			if err != nil {
+				return err
+			}
+
+			namespace, err := getNamespace(*o.configFlags.Namespace)
+			if err != nil {
+				return err
+			}
+
+			if len(nodeSelector) != 0 && len(args) != 0 {
+				return fmt.Errorf("it's not allowed to use the node-selector and pass nodes")
+			}
+
+			if len(nodeSelector) != 0 {
+				nodes, err := getNodes(kubeClient, nodeSelector)
+				if err != nil {
+					return err
+				}
+
+				return evacuateNode(kubeClient, cluster, nodes, namespace, withExclusion, force)
+			}
+
+			return evacuateNode(kubeClient, cluster, args, namespace, withExclusion, force)
+		},
+		Example: `
+# Evacuate all instances for a cluster in the current namespace that are hosted on node-1
+kubectl fdb evacuate -c cluster node-1
+
+# Evacuate all instances for a cluster in the default namespace that are hosted on node-1
+kubectl fdb evacuate -n default -c cluster node-1
+
+# Evacuate all instances for a cluster in the current namespace that are hosted on nodes with the labels machine=a,disk=fast
+kubectl fdb evacuate -c cluster --node-selector machine=a,disk=fast
+`,
+	}
+
+	cmd.Flags().StringP("fdb-cluster", "c", "", "evacuate instance(s) from the provided cluster.")
+	cmd.Flags().StringToStringVarP(&nodeSelectors, "node-selector", "", nil, "node-selector got all nodes that should be evacuated.")
```

Comment:
I'm not sure that "got" is the right word here.

Reply:
Definitely not 🤔